src/transformer/descriptor/typeNode/convertNodeToTypeNode.ts

```diff
@@ -0,0 +1,49 @@
+import type * as ts from 'typescript';
+import { core } from '../../core/core';
+import { TransformerLogger } from '../../logger/transformerLogger';
+import { GetNullDescriptor } from '../null/null';
+
+export function convertNodeToTypeNode(
+  node: ts.Node
+): ts.TypeNode | ts.Expression {
+  const typeChecker: ts.TypeChecker = core.typeChecker;
+
+  const typeAtLocation: ts.Type = typeChecker.getTypeAtLocation(node);
+
+  const typeNode: ts.TypeNode | undefined = typeChecker.typeToTypeNode(
+    typeAtLocation,
+    undefined,
+    undefined
+  );
+
+  if (!typeNode) {
+    TransformerLogger().typeCannotBeChecked(node);
+    return GetNullDescriptor();
+  }
+
+  if (!canTypeBeChecked(typeNode)) {
```

Comment:
if the type can or cannot be checked the result is the same, is this useful?

Reply:
Probably I need to return a NullDescriptor.  :) thanks! I'll see if I can write a test.

Reply:
Done, cannot find a test